code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,916 @@
+	sparks.set_up(3, 3, loc)
+	sparks.start()
+	shake_the(src)
+	if(prob(30))
+		say(pick("Fu-ue-ewe-eweu-u-uck!", "A-a-ah! Sto-op! Stop it pl-pleasuew...", "Go-o-o-od God-d-dpf!", "BZE-EW-EWQ! He-e-el-l-el!"))
+	addtimer(CALLBACK(src, .proc/malfunction_result), 2 SECONDS)
+
+
+//It's second proc, result of our malfunction
+/mob/living/simple_animal/hostile/hivemind/proc/malfunction_result()
+	if(prob(malfunction_chance))
+		apply_damage(rand(10, 25), BURN)
+
+
+//sometimes, players use closets, to staff mobs into it
+//and it's works pretty good, you just weld it and that's all
+//but not this time
+/mob/living/simple_animal/hostile/hivemind/proc/closet_interaction()
+	if(mob_size >= MOB_MEDIUM)
+		var/obj/structure/closet/closed_closet = loc
+		if(closed_closet && istype(closed_closet))
+			closed_closet.open(src)
+
+
+/mob/living/simple_animal/hostile/hivemind/say()
+	..()
+	playsound(src.loc, pick('sound/machines/robots/robot_talk_heavy1.ogg',
+							'sound/machines/robots/robot_talk_heavy2.ogg',
+							'sound/machines/robots/robot_talk_heavy3.ogg',
+							'sound/machines/robots/robot_talk_heavy4.ogg',
+							'sound/machines/robots/robot_talk_light1.ogg',
+							'sound/machines/robots/robot_talk_light2.ogg',
+							'sound/machines/robots/robot_talk_light3.ogg',
+							'sound/machines/robots/robot_talk_light4.ogg',
+							'sound/machines/robots/robot_talk_light5.ogg',
+							), 50, 1)
+
+
+/mob/living/simple_animal/hostile/hivemind/Life()
+	//to prevent life ticking at dead mobs and repeatedly calling death(), an issue, i think
+	if(stat != CONSCIOUS)
+		return
+
+	speak()
+
+	if(malfunction_chance && prob(malfunction_chance))
+		mulfunction()
+
+	closet_interaction()
+
+	..()
+
+
+/mob/living/simple_animal/hostile/hivemind/AttackingTarget()
+	..()
+
+/mob/living/simple_animal/hostile/hivemind/proc/speak()
+	if(!client && speak_chance && prob(speak_chance) && speak.len)
+		if(target_mob && target_speak.len)
+			say(pick(target_speak))
+		else
+			say(pick(speak))
+
+
+//no, they not instantly die by emp, but it's raise malfunction chance
+//due to nature of malfunction, they just burn to death sometimes
+/mob/living/simple_animal/hostile/hivemind/emp_act(severity)
+	switch(severity)
+		if(1)
+			if(malfunction_chance < 20)
+				malfunction_chance = 20
+		if(2)
+			if(malfunction_chance < 30)
+				malfunction_chance = 30
+
+
+/mob/living/simple_animal/hostile/hivemind/death()
+	if(master) //for spawnable mobs
+		master.spawned_creatures.Remove(src)
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+
+
+
+///life's///////////////////////////////////////////////////
+////////////////////////////////RESSURECTION///////////////
+///////////////////////////////////////////////go on//////
+
+
+//these guys is appears from bodies, and takes corpses appearence
+/mob/living/simple_animal/hostile/hivemind/ressurected
+	name = "ressurected creature"
```

Comment:
The right word is "resurrected".

Overall, the text quality in descs is meh, but I expect that to be improved later.